ocaml/xapi/vgpuops.ml

```diff
@@ -140,7 +140,7 @@ let add_vgpus_to_vm ~__context host vm vgpus vgpu_manual_setup =
       if not vgpu_manual_setup then
         let pgpu = allocate_vgpu_to_gpu ~__context vm host vgpu in
```

Comment:
There could still be a race: two or more threads at the same time could reach this line and call `allocate_vgpu_to_gpu` and be allocated to a pgpu that has only one free VF, so then one of them will fail with `"No free virtual function found"` from `get_free_virtual_function` even if there are plenty of free VFs on other pgpus.